### PR TITLE
Main nws14 read option

### PR DESCRIPTION
**Pull request by dwirasae**
# Description

Resurrect an approach reading NetCdf nws14 met files using all compute core to simultaneously read in data. This option is activated through specifying a namelist:
    
&WindGrib2NetCdf read_NWS14_NetCdf_using_core_0 = .false.

in fort.15. If this namelist is omitted or set to .true., a core 0 will read in met data and MPI broadcasts to all other compute cores. 

Faster runtime using this approach seen in some HPC systems is a primary reason to bring it back (as an option).   

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Bug fix with breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Update to existing feature to add new functionality
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (feature that would cause existing functionality to not work as expected)

# Issue Number

# How Has This Been Tested?

The changes were tested on a few small test problems (Shinnecock, small global runs).

# Relevant Publications (if applicable)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [ ] My code is up-to-date and tested with changes from the latest version of `main`

## If any of the above are not checked, please explain why:

# Further comments

### CODE REVIEW

File: src/wind.F [hunk 1]
```diff
@@ -9685,6 +9752,41 @@ END SUBROUTINE ArnoldSchwarzenegger
...
+        use global, only: NWS, NCICE
+        USE SIZES, ONLY : MYPROC
+    
+#ifdef ADCNETCDF
+        use netcdf
+#endif
+
+        IMPLICIT NONE
+
+#ifdef DATATIME
```
Comment (zcobell) [inline src/wind.F:9766]: Should this be `DATETIME`?

File: src/wind.F [hunk 2]
```diff
@@ -8925,22 +8928,43 @@ SUBROUTINE NWS14SET_FILENAMES()
...
-#ifdef CMPI
-         IF (MYPROC.EQ.0) THEN
-#endif
-            call Check_err(NF90_OPEN(Pfile,nf90_nowrite,PfileNCID))
-            call Check_err(NF90_OPEN(Wfile,nf90_nowrite,WfileNCID))
-            call Check_err(NF90_OPEN(Wfile1,nf90_nowrite,Wfile1NCID))
-            IF (NCICE.EQ.14) THEN
-               call Check_err(NF90_OPEN(Cfile,nf90_nowrite,CfileNCID))
-            ENDIF
+!#ifdef ADCNETCDF ! BEG ADCNETCDF
```
Comment (zcobell) [inline src/wind.F:8931]: Please remove this commented out. block

**Review by zcobell — CHANGES_REQUESTED** [review]
Hi @dwirasae - Looks good. Couple minor changes and we'll get this merged

File: src/wind.F (hunk 1 above)
Comment (dwirasae) [reply src/wind.F:9766]: Yes.

File: src/wind.F (hunk 2 above)
Comment (dwirasae) [reply src/wind.F:8931]: Will do.

**Comment by zcobell** [comment]
Looks good. Thanks @dwirasae! I'm going to compress these into a single commit and merge. Much appreciated.

**Review by zcobell — APPROVED** [review]

**Pull request merged**